--- Python-Docs-2.4/./howto/TODO	1970-01-01 09:00:00.000000000 +0900
+++ Python-Docs-2.5/./howto/TODO	2005-08-31 23:43:10.000000000 +0900
@@ -0,0 +1,13 @@
+
+Short-term tasks:
+  Quick revision pass to make HOWTOs match the current state of Python:
+curses doanddont regex sockets sorting
+
+Medium-term tasks:
+ Revisit the regex howto.  
+	* Add exercises with answers for each section
+	* More examples?
+
+Long-term tasks:
+ Integrate with other Python docs?
+
